### PR TITLE
Fusion to support extra add after QK Matmul without mask

### PR DESCRIPTION
**Pull request by apsonawane**
### Description

Fusion required for layoutxlm model. This model is based on BERT model but has an extra Add after the QK Matmul but the functionality is different than the existing one.

qk

### CODE REVIEW

File: onnxruntime/python/tools/transformers/fusion_attention.py [hunk 1]
```diff
@@ -900,13 +901,17 @@ def create_attention_node(
                 past_kv = self.concat_kv(past_k, past_v)
                 attention_inputs.append(past_kv)
 
-            if add_qk_str is not None:
+            if add_qk_str is not None and has_relative_pos_bias is None:
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/fusion_attention.py:904]: has_relative_pos_bias is boolean. Why compare it with None?

File: onnxruntime/python/tools/transformers/fusion_attention.py [hunk 2]
```diff
@@ -1040,12 +1045,14 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
...
         is_distill_add = False
         is_no_mask_attention = False
+        has_rel_pos_bias = False
         qk_paths = {
             "path1": (["Softmax", "Add", "Div", "MatMul"], [0, 0, None, 0]),
             "path2": (["Softmax", "Add", "Mul", "MatMul"], [0, 0, None, 0]),
             "path3": (["Softmax", "Where", "MatMul", "Div"], [0, 0, 2, 0]),
             "path4": (["Softmax", "Add", "Where", "MatMul"], [0, 0, 0, 2]),
             "path5": (["Softmax", "Div", "MatMul"], [0, 0, 0]),
+            "path6": (["Softmax", "Add", "Add", "Div", "MatMul"], [0, 0, None, 0, 0]),
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/fusion_attention.py:1055]: This pattern is different from the image in description:
image

File: onnxruntime/python/tools/transformers/fusion_attention.py [hunk 3]
```diff
@@ -1133,6 +1145,13 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
                     return
         elif is_no_mask_attention:
             pass
+        elif has_rel_pos_bias:
+            mask_nodes = self.model.match_parent_path(
+                add_qk,
+                ["Mul", "Sub", "Cast", "Unsqueeze", "Unsqueeze"],
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/fusion_attention.py:1151]: Check also which axes used in unsqueeze? Does add_qk support such shape (looks like need broadcasting in kernel)?

**Comment by tianleiwu** [comment]
Please add a test

File: onnxruntime/python/tools/transformers/fusion_attention.py (hunk 1 above)
Comment (apsonawane) [reply onnxruntime/python/tools/transformers/fusion_attention.py:904]: Yeah will update it

File: onnxruntime/python/tools/transformers/fusion_attention.py (hunk 2 above)
Comment (apsonawane) [reply onnxruntime/python/tools/transformers/fusion_attention.py:1055]: Yes

**Comment by apsonawane** [comment]
No longer required

**Pull request closed without merging**